### PR TITLE
feat(dynamic-plugins): make mountpoints and layout declarative

### PR DESCRIPTION
**Pull request by tumido**
Use dynamic frontend plugins across the app:

1. Dynamic routes support home `/` override. You can define additional routes via `dynamicPlugins.frontend.dynamicRoutes` - these can't already exists in the app with 1 exception - `/`. This allows you to override the home page with your own plugin/component.

2. This change makes `dynamicPlugins.frontend.mountPoints` generic and declarative:

   Mountpoint now support following names/types:

   - Allow passing `*/context` mountpoints for React context
   - Allow passing `*/cards` for Card components (with layout)

   Mountpoint configs:

   - `entity.page.overview`
   - `entity.page.topology`
   - `entity.page.issues`
   - `entity.page.pull-requests`
   - `entity.page.ci`
   - `entity.page.cd`
   - `entity.page.kubernetes`
   - `entity.page.tekton`
   - `entity.page.image-registry`
   - `entity.page.monitoring`
   - `entity.page.lighthouse`
   - `entity.page.api`
   - `entity.page.dependencies`
   - `entity.page.docs`
   - `entity.page.definition`
   - `entity.page.diagram`

   Mountpoints support following configuration:

   - `layout` for layout features that propagates to <Box sx=... /> allowing users to use CSS properties gridColumnStart including responsiveness queries etc. (mui.com/system/ getting-started/the-sx-prop)
   - `if` for EntitySwitch.Case if=... - allows allOf|anyOf|oneOf conditionals with isKind|isType|hasAnnotation builtin methods or code imports via Scalprum (direct string reference)
   - `props` to pass additional props to the mounted component

   Current limitations of the dynamic frontend plugins:

   Allows you to mount to existing mountPoints only. You're unable to create additional tabs for example. (will be addressed in a follow up PR)

https://github.com/janus-idp/backstage-showcase/assets/7453394/956de326-8964-4bde-b974-5645ee55efbd

**Known issues:**

If a frontend package requires config schema providing `visibility: frontend` annotations, it will fail, since we currently have no means of aggregating it. This renders plugins like Lighthouse unusable at the moment.

## Which issue(s) does this PR fix

Part of #442 
Requires: #699 #683

## PR acceptance criteria

Please make sure that the following steps are complete:

- [ ] GitHub Actions are completed and successful
- [ ] Unit Tests are updated and passing
- [ ] E2E Tests are updated and passing
- [ ] Documentation is updated if necessary (requirement for new features)
- [ ] Add a screenshot if the change is UX/UI related

### CODE REVIEW

File: dynamic-plugins/imports/package.json [hunk 1]
```diff
@@ -15,11 +15,11 @@
...
-    "@janus-idp/backstage-plugin-ocm": "3.3.2",
-    "@janus-idp/backstage-plugin-quay": "1.4.3",
-    "@janus-idp/backstage-plugin-tekton": "3.1.1",
-    "@janus-idp/backstage-plugin-topology": "1.16.1"
+    "@janus-idp/backstage-plugin-jfrog-artifactory": "1.2.3",
+    "@janus-idp/backstage-plugin-nexus-repository-manager": "1.4.3",
+    "@janus-idp/backstage-plugin-ocm": "3.3.3",
+    "@janus-idp/backstage-plugin-quay": "1.4.5",
+    "@janus-idp/backstage-plugin-tekton": "3.1.2",
+    "@janus-idp/backstage-plugin-topology": "1.16.3"
```
Comment (invincibleJai) [inline dynamic-plugins/imports/package.json:23]: Not sure if this is valid but as we have imports here so shall we remove these from app -> package.json?

https://github.com/janus-idp/backstage-showcase/blob/main/packages/app/package.json#L57-L62

Comment (tumido) [reply dynamic-plugins/imports/package.json:23]: yup, we can safely do that now :) Good point!

File: plugins/scalprum-backend/src/service/router.ts [hunk 2]
```diff
@@ -24,6 +24,7 @@ export async function createRouter(options: RouterOptions): Promise<Router> {
 
   const availablePackages = pluginManager.availablePackages;
   const plugins = pluginManager.plugins;
+  console.log(plugins);
```
Comment (gashcrumb) [inline plugins/scalprum-backend/src/service/router.ts:27]: Is this console.log necessary?

**Review by davidfestal — COMMENTED** [review]
Still testing this with the Helm chart, and having errors.

**Review by gashcrumb — APPROVED** [review]

**Review by davidfestal — APPROVED** [review]

**Review by schultzp2020 — APPROVED** [review]
LGTM

**Pull request merged**